viam-cartographer/cartographer_patches/carto.patch

```diff
@@ -0,0 +1,304 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index ef2fcb3..ba98945 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1,33 +1,17 @@
+-# Copyright 2016 The Cartographer Authors
+-#
+-# Licensed under the Apache License, Version 2.0 (the "License");
+-# you may not use this file except in compliance with the License.
+-# You may obtain a copy of the License at
+-#
+-#      http://www.apache.org/licenses/LICENSE-2.0
+-#
+-# Unless required by applicable law or agreed to in writing, software
+-# distributed under the License is distributed on an "AS IS" BASIS,
+-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-# See the License for the specific language governing permissions and
+-# limitations under the License.
+-
++# viam-patched version
+ cmake_minimum_required(VERSION 3.2)
+ 
+ project(cartographer)
++add_definitions(-DABSL_LEGACY_THREAD_ANNOTATIONS)
+ 
+ set(CARTOGRAPHER_MAJOR_VERSION 1)
+ set(CARTOGRAPHER_MINOR_VERSION 0)
+ set(CARTOGRAPHER_PATCH_VERSION 0)
+ set(CARTOGRAPHER_VERSION ${CARTOGRAPHER_MAJOR_VERSION}.${CARTOGRAPHER_MINOR_VERSION}.${CARTOGRAPHER_PATCH_VERSION})
+ set(CARTOGRAPHER_SOVERSION ${CARTOGRAPHER_MAJOR_VERSION}.${CARTOGRAPHER_MINOR_VERSION})
+-option(BUILD_GRPC "build Cartographer gRPC support" false)
+-set(CARTOGRAPHER_HAS_GRPC ${BUILD_GRPC})
+-option(BUILD_PROMETHEUS "build Prometheus monitoring support" false)
+ 
+ include("${PROJECT_SOURCE_DIR}/cmake/functions.cmake")
+ google_initialize_cartographer_project()
+-google_enable_testing()
+ 
+ find_package(absl REQUIRED)
+ set(BOOST_COMPONENTS iostreams)
+@@ -38,21 +22,14 @@ endif()
+ find_package(Boost REQUIRED COMPONENTS ${BOOST_COMPONENTS})
+ find_package(Ceres REQUIRED COMPONENTS SuiteSparse)
+ find_package(Eigen3 REQUIRED)
+-find_package(LuaGoogle REQUIRED)
++find_package(Lua 5.2 REQUIRED)
++
+ if(WIN32)
+   # On Windows, Protobuf is incorrectly found by the bundled CMake module, so prefer native CMake config.
+   set(protobuf_MODULE_COMPATIBLE TRUE CACHE INTERNAL "")
+   find_package(Protobuf 3.0.0 CONFIG)
+ else()
+-  find_package(Protobuf 3.0.0 REQUIRED)
+-endif()
+-
+-if (${BUILD_GRPC})
+-  find_package(async_grpc REQUIRED)
+-endif()
+-
+-if(${BUILD_PROMETHEUS})
+-  find_package( ZLIB REQUIRED )
++  find_package(Protobuf 3.4.0 REQUIRED)
+ endif()
+ 
+ include(FindPkgConfig)
+@@ -62,12 +39,6 @@ else()
+   find_library(CAIRO_LIBRARIES cairo)
+ endif()
+ 
+-# Only build the documentation if we can find Sphinx.
+-find_package(Sphinx)
+-if(SPHINX_FOUND)
+-  add_subdirectory("docs")
+-endif()
+-
+ # Install catkin package.xml
+ install(FILES package.xml DESTINATION share/cartographer)
+ 
+@@ -80,9 +51,6 @@ install(DIRECTORY cmake DESTINATION share/cartographer/)
+ 
+ file(GLOB_RECURSE ALL_LIBRARY_HDRS "cartographer/*.h")
+ file(GLOB_RECURSE ALL_LIBRARY_SRCS "cartographer/*.cc")
+-file(GLOB_RECURSE TEST_LIBRARY_HDRS "cartographer/fake_*.h" "cartographer/*test_helpers*.h" "cartographer/mock_*.h")
```

Comment:
I don't know. It wasn't my change. 90% of this file was already here. I just turned it from a file that got copied on top of the submodule, to a patch against it, since I also needed to modify another file.

A patch is just easier, as it can be any number of files/changes, vs having to explicitly copy multiple files to specific locations.

The only real changes I made to the inner CMakeLists.txt were removing the CERES_INCLUDE_DIRS bit, adding the comment at the top, and adding the conditional link libraries for absl::log_internal_check_op